tests/test_gui.py

```diff
@@ -11,19 +11,20 @@ def close_figures_on_teardown():
     plt.close("all")
 
 
-@pytest.mark.parametrize('root', ["./tests/example_data/pc",
-                                  "./tests/example_data/mn",
-                                  skipif_no_h5py("./tests/example_data/un"),
-                                  "./tests/example_data/nf"])
+@pytest.mark.parametrize('root', [
+    "./tests/example_data/pc",
+    "./tests/example_data/mn",
+    skipif_no_h5py("./tests/example_data/un"),
+    "./tests/example_data/nf",
+    "./tests/example_data/dnest4"])
 def test_gui(root):
     samples = read_chains(root)
     plotter = samples.gui()
 
     # Type buttons
-    plotter.type.buttons.set_active(1)
-    assert plotter.type() == 'posterior'
-    plotter.type.buttons.set_active(0)
-    assert plotter.type() == 'live'
+    for i, plot_type in enumerate(samples.plot_types()):
+        plotter.type.buttons.set_active(i)
+        assert plotter.type() == plot_type
```

Comment:
This change, although one line shorter, is not actually better. For unit tests it is better to repeat and make sure any issue can be pinned to a specific line. Loops leave unclear at which iteration in the loop an issue occured. So better to repeat in tests.

That said, excessive repeats are of course annoying to maintain. But the better way of handling that is with pytest's `parametrize` options.